src/ranch_acceptors_sup.erl

```diff
@@ -25,28 +25,52 @@ start_link(Ref, NumAcceptors, Transport) ->
 
 init([Ref, NumAcceptors, Transport]) ->
 	TransOpts = ranch_server:get_transport_options(Ref),
-	Logger = maps:get(logger, TransOpts, error_logger),
+	NumListenSockets = maps:get(num_listen_sockets, TransOpts, 1),
 	SocketOpts = maps:get(socket_opts, TransOpts, []),
+	Logger = maps:get(logger, TransOpts, error_logger),
 	%% We temporarily put the logger in the process dictionary
 	%% so that it can be used from ranch:filter_options. The
 	%% interface as it currently is does not allow passing it
 	%% down otherwise.
 	put(logger, Logger),
-	LSocket = case Transport:listen(SocketOpts) of
+	LSockets = start_listen_sockets(Ref, NumListenSockets, Transport, SocketOpts, Logger),
+	erase(logger),
+	[{_, BaseSocket}|_] = LSockets,
+	{ok, Addr} = Transport:sockname(BaseSocket),
```

Comment:
You already call this function in start_listen_sockets, no need to call it twice. Just call ranch_server:set_addr in start_listen_sockets.

Reply:
Or return the Addr from start_listen_sockets and then call ranch_server:set_addr.